src/orbit_determination/gauss_method.jl

```diff
@@ -114,125 +118,142 @@ end
 
 Lagrange polynomial to be solved during Gauss method.
 """
-lagrange(x::T, a::U, b::U, c::U) where {T, U <: Number} = x^8 + a*x^6 + b*x^3 + c
+function lagrange(x::T, a::U, b::U, c::U) where {T, U <: Number}
+    # Evaluate via Horner's method
+    x2 = x * x
+    x3 = x2 * x
+    return c + x3 * (b + x3 * (a + x2))
+end
 
 @doc raw"""
     lagrange_derivative(x::T, a::T, b::T) where {T <: Number}
 
 Derivative of Lagrange polynomial to be solved during Gauss method.
 """
-lagrange_derivative(x::T, a::U, b::U) where {T, U <: Number} = 8*x^7 + 6*a*x^5 + 3*b*x^2
+function lagrange_derivative(x::T, a::U, b::U) where {T, U <: Number}
+    # Evaluate via Horner's method
+    x2 = x * x
+    x3 = x2 * x
+    return x2 * (3*b + x3 * (6*a + 8*x2))
+end
 
 # TO DO: Allow to control interval over which to look for solutions
-# Currently we look between the radius of the Sun ((0.00465047 au) and the radius of the Solar System (40 au)
+# Currently we look between the radius of the Sun (∼0.00465047 AU) and
+#  the radius of the Solar System (∼40 AU)
 
 @doc raw"""
-    solve_lagrange(a::T, b::T, c::T; niter::Int = niter) where {T <: Real}
-    solve_lagrange(a::TaylorN{T}, b::TaylorN{T}, c::TaylorN{T}; niter::Int = niter) where {T <: Real}
+    solve_lagrange(a::T, b::T, c::T; niter::Int = 5) where {T <: Real}
+    solve_lagrange(a::TaylorN{T}, b::TaylorN{T}, c::TaylorN{T}; niter::Int = 5) where {T <: Real}
 
 Solve Lagrange polynomial.
 
 See also [`lagrange`](@ref).
 """
 function solve_lagrange(a::T, b::T, c::T; niter::Int = 5) where {T <: Real}
-    sol = roots(x -> lagrange(x, a, b, c), Interval(0.00465047, 40))
-    return mid.(interval.(sol)), getfield.(sol, :status)
+    return find_zeros(x -> lagrange(x, a, b, c), 0.00465047, 40.0)
```

Comment:
I suggest to add the boundaries to the search intervals as keyword arguments to `solve_lagrange` instead of having them set manually in the function body

Reply:
Solved by https://github.com/PerezHz/NEOs.jl/pull/31/commits/b67b85ec7f3e4b5d0bf443cce1388194d6fbf464.